### DISCUSSION TITLE
Error When Running React App on Github Pages - Uncaught SyntaxError: Unexpected token '<'

### DISCUSSION THREAD

**Discussion by tim19889** [community/community#52329] [Other Feature Feedback, Questions, & Ideas] [answered]
### Select Topic Area

Bug

### Body

Hi All!

I'm an aspiring web developer who recently completed Free Code Camp's JS, CSS, HTML, and Front End Development Libraries courses. I've created my first Github Pages website and am attempting to move all the HTML, CSS, and JS/React code for the projects I built in Codepen to Github Pages. I want to create a portfolio website of my projects. Right now I haven't done anything with the homepage. I am just trying to upload all of the relevant HTML, CSS, and JS files.

Any app I've created with vanilla JS I can upload along with the HTML and CSS files and it works fine. However, any apps where the JS file contains React does not load in the browser. Instead, when I open up the console I receive this message, "Uncaught SyntaxError: Unexpected token '<' ".

Here is the HTML page of one of my React apps.

https://tim19889.github.io/cashregister.html 

Here is the repository which contains the cashregister.html, cashregister.css, and cashregister.js files.

https://github.com/tim19889/tim19889.github.io

Any tips for troubleshooting this further would be greatly appreciated. I have researched the issue but haven't been able to figure out the solution yet unfortunately.

**Comment by tim19889** [comment]
Bump. Any ideas?

**Comment by iaruso** [comment]
Hope this helps, but the only way I could make it work is by doing this:

1. Add the following code in html right after div#root:

`<script src="https://unpkg.com/babel-standalone@6/babel.min.js"></script>`
`<script src="https://cdnjs.cloudflare.com/ajax/libs/react/16.6.3/umd/react.production.min.js"></script>`
`<script src="https://cdnjs.cloudflare.com/ajax/libs/react-dom/16.6.3/umd/react-dom.production.min.js"></script>`
    
_This versions work, but if you want, check for newer ones._

2. Add type="text/babel" like this:

`<script type="text/babel" src="cashregister.js" type="module"></script>`

3. In the javascript file remove all imports:

`import React, { useState } from "https://cdn.skypack.dev/react";`
`import ReactDOM from "https://cdn.skypack.dev/react-dom";`
`import useStateInCustomProperties from "https://cdn.skypack.dev/use-state-in-custom-properties";`

It worked for me :), but you will have to repeat this for each html and js file you have, I personally recommend to use npm and learn how to "properly" use React because you have a lot of pages in that repository.

![image](https://user-images.githubusercontent.com/90287515/231148490-81d52b6e-68c7-46b2-91ba-6503b9a40293.png)

**Reply by tim19889** [reply]
Thanks @iaruso ! I tried your suggestions but unfortunately it still didn't display for me. Possibly due to the way my browser is set up. Anyway, before I read your reply, I actually did get started deploying with NPM. I managed to get another react app at "https://tim19889.github.io/mywordle" to run when running "npm start" in the local test environment, but when I run "npm run build" to actually deploy it I'm getting a "The target environment doesn't support dynamic import() syntax so it's not possible to use external type 'module' within a script" after trying to build react app using "npm run build"" error message.

Would this be because of my imports at the top of the React code in my app.js file? If so, how am I supposed to import?

**Reply by iaruso** [reply, marked as answer]
No problem, it's just better and easier to use npm. In your case, just follow these steps:

1. In your repository folder terminal, enter the following commands:
`npm i react` and then `npm i react-dom` to install those packages

2. Update your import statements as follows:
`import React, { useState } from "https://cdn.skypack.dev/react";` to `import React, { useState } from "react";`
`import ReactDOM from "https://cdn.skypack.dev/react-dom";` to `import ReactDOM from "react-dom";`
and remove the `import useStateInCustomProperties from "https://cdn.skypack.dev/use-state-in-custom-properties";`

    _Please avoid using CDN imports in JavaScript files (when using React). Instead, use npm packages (npm install package-name)._

If this solution works for you, please mark it as answered so that other users can benefit from it.

**Reply by tim19889** [reply]
That worked like a charm, thank you iaruso!
